SetupDataPkg/Tools/KnobService.py

```diff
@@ -1172,6 +1172,21 @@ def generate_profiles(schema, profile_header_path, profile_paths, efi_type, prof
                 out.write(get_spacing_string(efi_type) + '"' + profile_name + '",' + get_line_ending(efi_type))
             out.write("};" + get_line_ending(efi_type))
             out.write(get_line_ending(efi_type))
+            if profile_ids is not None:
+                ids_list = profile_ids.split(",")
+            else:
+                # If not specified, the indices will be the default profile ids
+                ids_list = [format(i, '02x') for i in range(len(profile_paths))]
+
+            out.write(get_line_ending(efi_type))
+            out.write(get_type_string("uint8_t", efi_type) + " g{}[PROFILE_COUNT]".format(
+                naming_convention_filter("_profile_flavor_ids", False, efi_type)) + " = {"
+            )
+            out.write(get_line_ending(efi_type))
+            for profile_id in ids_list:
+                out.write(get_spacing_string(efi_type) + '0x' + profile_id + ',' + get_line_ending(efi_type))

```

Comment:
This seems to take the incoming string as is? So it means that there is some requirement on the input format? Can we please:

1. Update the argument print message for `-pid` for users, so that we know what input string they should give? (i.e. decimal, hex, including `0x`, etc)
2. Once we clarify the above point, we should actually convert the `profile_id` here from string to integer and convert it back into autogen header strings. This way we have better sanity on the input and use case handling?

Reply:
Updated code as per the inputs.